pkg/pricer/pricer_test.go

```diff
@@ -0,0 +1,373 @@
+	if !reflect.DeepEqual(table3, getTable3) {
+		t.Fatalf("returned table does not match, got %+v expected %+v", getTable3, table3)
+	}
+
+	getTable4 := pricer.PriceTable()
+	if !reflect.DeepEqual(table4, getTable4) {
+		t.Fatalf("returned table does not match, got %+v expected %+v", getTable4, table4)
+	}
+
+}
+
+func TestPeerPrice(t *testing.T) {
+
+	logger := logging.New(ioutil.Discard, 0)
+
+	store := mock.NewStateStore()
+	defer store.Close()
+
+	overlay := swarm.MustParseHexAddress("0000617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700")
+
+	peer := swarm.MustParseHexAddress("e5ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700")
+	chunksByPOToPeer := []swarm.Address{
+		swarm.MustParseHexAddress("05ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("95ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("c5ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("f0ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("e6ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+	}
+
+	pricer := pricer.New(logger, store, overlay, 10)
+
+	kad := mockkad.NewMockKademlia(mockkad.WithDepth(3))
+	pricer.SetKademlia(kad)
+
+	peerTable := []uint64{55, 45, 35, 25, 15}
+	err := pricer.NotifyPriceTable(peer, peerTable)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i, ch := range chunksByPOToPeer {
+
+		getPrice := pricer.PeerPrice(peer, ch)
+		if getPrice != peerTable[i] {
+			t.Fatalf("unexpected PeerPrice, got %v expected %v", getPrice, peerTable[i])
+		}
+	}
+
+	neighborhoodPeer := swarm.MustParseHexAddress("00ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700")
+	neighborhoodPeerTable := []uint64{36, 26, 16, 6}
+
+	err = pricer.NotifyPriceTable(neighborhoodPeer, neighborhoodPeerTable)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	chunksByPOToNeighborhoodPeer := []swarm.Address{
+		swarm.MustParseHexAddress("95ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("55ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("35ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+		swarm.MustParseHexAddress("10ef617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700"),
+	}
+
+	neighborhoodPeerExpectedPrices := []uint64{36, 26, 16, 0}
+
+	for i, ch := range chunksByPOToNeighborhoodPeer {
+		getPrice := pricer.PeerPrice(neighborhoodPeer, ch)
+		if getPrice != neighborhoodPeerExpectedPrices[i] {
+			t.Fatalf("unexpected PeerPrice, got %v expected %v", getPrice, neighborhoodPeerExpectedPrices[i])
+		}
+	}
+
+}
+
+func TestPriceForPeer(t *testing.T) {
+
+	logger := logging.New(ioutil.Discard, 0)
+
+	store := mock.NewStateStore()
+	defer store.Close()
+
+	peer := swarm.MustParseHexAddress("0000617cadab2af7fff48b16b52953487a22589362e34efdd7e0d4ba628ad700")
```

Comment:
you actually don't need to have full length addresses in these tests, since `swarm.Address` does not (yet) enforce any size limit or minimum requirement. it can improve readability of the tests